force-app/main/default/classes/RecurrenceRule.cls

```diff
@@ -73,64 +73,141 @@ public with sharing class RecurrenceRule {
+        return count;
     }
 
     public List<Day> getDays() {
-        return this.days;
+        return days;
     }
 
     public String build() {
-        // TODO: build the rule string from the RecurrenceRule object
-        return this.rRule;
-    }
+        List<String> ruleParts = new List<String>();
+
+        // FREQ (required)
+        ruleParts.add('FREQ=' + freq);
+
+        // INTERVAL
+        if (interval != 1) {
+            ruleParts.add('INTERVAL=' + interval);
+        }
 
-    private void parse() {
-        // TODO: parse the rule string into a RecurrenceRule object
+        // UNTIL
+        if (endDate != null) {
+            String endDateString = DateTime.newInstanceGMT(
+                    endDate,
+                    Time.newInstance(0, 0, 0, 0)
+                )
+                .formatGmt('yyyyMMdd');
+
+            ruleParts.add('UNTIL=' + endDateString + 'T000000Z');
+        }
+
+        // COUNT
+        if (count != null) {
+            ruleParts.add('COUNT=' + count);
+        }
+
+        // BYDAY
+        if (days != null && days.size() > 0) {
+            List<String> dayStrings = new List<String>();
+            for (Day day : days) {
+                dayStrings.add(day.name());
+            }
+            ruleParts.add('BYDAY=' + String.join(dayStrings, ','));
+        }
+
+        rRule = String.join(ruleParts, ';');
+        return rRule;
+    }
+
+    private void parseAndValidate(String rRuleString) {
+        List<String> ruleParts = rRuleString.split(';');
+
+        for (String part : ruleParts) {
+            List<String> thisPart = part.split('=');
+
+            if (!thisPart.isEmpty()) {
+                String thisKey = thisPart[0].toUpperCase();
+                String thisValue = thisPart[1];
+
+                switch on thisKey {
+                    when 'FREQ' {
+                        setFreqFromString(thisValue);
+                    }
+                    when 'INTERVAL' {
+                        interval = Integer.valueOf(thisValue);
+                    }
+                    when 'UNTIL' {
+                        Integer year = Integer.valueOf(thisValue.left(4));
+                        Integer month = Integer.valueOf(thisValue.mid(4, 2));
+                        Integer day = Integer.valueOf(thisValue.mid(6, 2));
+                        endDate = Date.newInstance(year, month, day);
+                    }
+                    when 'COUNT' {
+                        count = Integer.valueOf(thisValue);
+                    }
+                    when 'BYDAY' {
+                        List<String> dayStrings = thisValue.split(',');
+                        days = new List<Day>();
+                        for (Day day : Day.values()) {
+                            if (dayStrings.contains(day.name())) {
+                                days.add(day);
+                            }
+                        }
+                    }
+                    when else {
```

Comment:
We might have to clarify here on rrule features and keys we've implemented, vs actual invalid keys.